src/app/modules/server-settings/pages/manage-settings/manage-settings.component.ts

```diff
@@ -127,6 +127,23 @@ export class ManageSettingsComponent implements OnInit, OnDestroy {
         this.destroy.complete();
     }
 
+    get nonProxyHost(): FormArray {
+        return this.settingsForm.get('proxyConfigurationForm.nonProxyHosts') as FormArray;
+    }
+
+    addNonProxyHost() {
+        const group = this.settingsForm.get('proxyConfigurationForm');
+        group['controls']['nonProxyHosts'].push(new FormControl(''));
```

Comment:
1. `FormControl` is missing [Validators.required](https://angular.io/guide/form-validation#built-in-validator-functions)
2. ```suggestion
           this.getNonProxyHosts().push(new FormControl(''));
    ```
3. missing `ChangeDetectorRef.detectChanges` call to causes style issues for the newly added element.